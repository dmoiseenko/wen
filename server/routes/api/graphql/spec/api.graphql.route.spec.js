const path = require('path');

const bootstrap = require('../../../../db/bootstrap/bootstrap');
const loadTextFile = require('../../../../../common/loadTextFile.js');
const login = require('../../utils/login');
const getClient = require('../../utils/getHttpClient');

const me = loadTextFile(path.join(__dirname, '../../../../../common/graphql/query/me.graphql'));
const notes = loadTextFile(path.join(
  __dirname,
  '../../../../../common/graphql/query/getAllNotes.graphql'
));
const addNote = loadTextFile(path.join(
  __dirname,
  '../../../../../common/graphql/mutation/addNote.graphql'
));


let Cookies;

beforeAll(async () => {
  Cookies = await login();
});

beforeEach(async () => {
  await bootstrap();
});


test('should process me query', async () => {
  const res = await getClient()
    .post('/api/graphql')
    .set('Cookie', Cookies)
    .send({
      query: me
    });

  expect(res.body).toMatchSnapshot();
});

test('should process notes query', async () => {
  const res = await getClient()
    .post('/api/graphql')
    .set('Cookie', Cookies)
    .send({
      query: notes
    });

  expect(res.body).toMatchSnapshot();
});

test('should process addNote Mutation', async () => {
  const res = await getClient()
    .post('/api/graphql')
    .set('Cookie', Cookies)
    .send({
      query: addNote,
      variables: {
        text: 'note text'
      }
    });

  expect(res.body).toMatchSnapshot();
});

