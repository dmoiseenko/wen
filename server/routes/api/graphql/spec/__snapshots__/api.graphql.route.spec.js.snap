// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should process addNote Mutation 1`] = `
Object {
  "data": Object {
    "addNote": Object {
      "id": 4,
      "text": "note text",
    },
  },
}
`;

exports[`should process me query 1`] = `
Object {
  "data": Object {
    "me": Object {
      "firstName": "John",
      "id": 1,
      "lastName": "Smith",
    },
  },
}
`;

exports[`should process notes query 1`] = `
Object {
  "data": Object {
    "notes": Array [
      Object {
        "id": 1,
        "text": "Well done is better than well said. Benjamin Franklin",
      },
      Object {
        "id": 2,
        "text": "Look deep into nature, and then you will understand everything better. Albert Einstein",
      },
      Object {
        "id": 3,
        "text": "Be yourself; everyone else is already taken. Oscar Wilde",
      },
    ],
  },
}
`;
